rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -193,12 +224,14 @@ def subscribe(self, client_id, topic, callback, msg_type=None):
 
         """
         if not topic in self._subscribers:
-            self._subscribers[topic] = MultiSubscriber(topic, msg_type)
+            self._subscribers[topic] = MultiSubscriber(
+                topic, client_id, callback, node_handle, msg_type=msg_type)
+        else:
+            self._subscribers[topic].subscribe(client_id, callback)
 
         if msg_type is not None:
             self._subscribers[topic].verify_type(msg_type)
 
-        self._subscribers[topic].subscribe(client_id, callback)
```

Comment:
In the old code this line triggered the 'hack' of adding and removing a callback, receiving a new message if the topic is latched. This happened with every call of this method, regardless if the `MultiSubscriber` was created in this call or not.

The patch proposed for latched topics is creating a new subscriber that processes the first message, updates the subscriptions and then is removed. But you don't need to do this with the first client id when `MultiSubscriber` is created, since the subscription will get any latched messages. That's why the new constructor initializes itself with a client id, and avoids creating a second subscription that uses `_new_sub_callback`.

I'm inclined for this approach as it is more efficient. Creating a `MultiSubscriber` without an initial client id has no use, as it has to be added afterwards anyway. Creating it with a client id avoids the need of creating a second subscription with a different callback just to end up using the first subscription originally created in the constructor.